### PR TITLE
Added unit testing section to the client README

### PR DESCRIPTION
**Pull request by Nerdinacan**
First draft at a set of instructions and introduction to client-side unit testing.

Feel free to throw fruit at it.

### CODE REVIEW

**Comment by martenson** [comment]
A leap forward. Thanks @Nerdinacan ! :shipit:

File: client/README.md [hunk 1]
```diff
@@ -77,6 +77,220 @@ do not apply site-wide.
...
+## The dirty secret about testing
+
+It's good to have tests, but testing code isn't really about testing at all,
+it's actually about software design. 
+
+Maintainable software is testable software. If you can run a unit test on your
+code, then that means you must have necessarily separated your code into
+testable units and you will have almost definitely written better, more modular,
+more easily manipulated code, and nobody will ever contemplate using git-blame
+to hunt you down.
```
Comment (mvdbeek) [inline client/README.md]:
```suggestion
figure out what went wrong.
```

File: client/README.md [hunk 2]
```diff
@@ -77,6 +77,220 @@ do not apply site-wide.
...
+
+### Clearly document the intent of your test
+
+Please remember that these tests are not *for* you. They're for the people who
+come after you. It will be a lot easier to modify, repair and upgrade your code
+if they can figure out what you were originally hoping to accomplish.
+
+Add a couple comments. Use variable names that mean something. Use the assertion
+failure description parameters when possible. I'm not sure what your mom told
+you, but your code is almost definitely not self-documenting.
```
Comment (mvdbeek) [inline client/README.md]: I think we should just drop the your mom statement, it'd still convey the same message.

**Review by mvdbeek — APPROVED** [review]
Awesome, we should have more of this!

File: client/README.md (hunk 2 above)
Comment (Nerdinacan) [reply client/README.md]: Removed! I didn't trust your mom anyway.

File: client/README.md [hunk 3]
```diff
@@ -77,6 +77,220 @@ do not apply site-wide.
...
-As mentioned above, `make client` will rebuild styles, as a part of the webpack
-build.  For iterative development, "Watch Mode" rebuilds as described above do
-include style changes.
+As mentioned above, `make client` will rebuild styles, as a part of the webpack build.  For iterative development, "Watch Mode" rebuilds as described above do include style changes.
+
+
+# Client-Side Unit Testing
+
+Galaxy's client is undergoing an extensive refactoring and modernizing process.
+As part of initiatve, we would like to request that all new client-side code
```
Comment (bgruening) [inline client/README.md]:
```suggestion
As part of this initiative, we would like to request that all new client-side code
```

File: client/README.md [hunk 4]
```diff
@@ -77,6 +77,220 @@ do not apply site-wide.
...
+## Testing ~~Suggestions~~ Obligations
+
+
+### Clearly document the intent of your test
+
+Please remember that these tests are not *for* you. They're for the people who
+come after you. It will be a lot easier to modify, repair and upgrade your code
+if they can figure out what you were originally hoping to accomplish.
+
+Add a couple comments. Use variable names that mean something. Use the assertion
```
Comment (bgruening) [inline client/README.md]:
```suggestion
Add a couple of comments. Use variable names that mean something. Use the assertion
```

File: client/README.md [hunk 5]
```diff
@@ -77,6 +77,220 @@ do not apply site-wide.
...
+## The dirty secret about testing
+
+It's good to have tests, but testing code isn't really about testing at all,
+it's actually about software design. 
+
+Maintainable software is testable software. If you can run a unit test on your
+code, then that means you must have necessarily separated your code into
+testable units and you will have almost definitely written better, more modular,
+more easily manipulated code, and nobody will ever contemplate using git-blame
+figure out what went wrong.
```
Comment (bgruening) [inline client/README.md]:
```suggestion
to figure out what went wrong.
```

**Comment by bgruening** [comment]
I learned a lot, thanks @Nerdinacan!

Ping @anuprulez @inkuzm

**Review by bgruening — APPROVED** [review]

**Pull request merged**